src/Private/CanonicalRequest.cs

```diff
@@ -45,6 +45,11 @@ public static class CanonicalRequest
             "trailer",
             "transfer-encoding",
             "upgrade",
+            "user-agent",
+            "via",
+            "x-forwarded-for",
+            "x-forwarded-port",
+            "x-forwarded-proto",
```

Comment:
API Gateway REST API (which uses CloudFront) always modifies "Via" and these three "X-Forwarded-*" headers. It also adds/modifies the "User-Agent" header if the incoming request doesn't have it or have it as an empty string.